tests/system-tests/rdscore/internal/rdscorecommon/ipvlan-validation.go

```diff
@@ -0,0 +1,391 @@
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to create deployment %s: %v", ipvlanDeploy21Name, err))
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Deployment %q created and is Ready in %q namespace",
+		deployTwo.Definition.Name, deployTwo.Definition.Namespace)
+
+	VerifyIPVLANConnectivityOnSameNode()
+}
+
+// VerifyIPVLANConnectivityOnSameNode verifies connectivity between workloads that use IPVLAN net.
+func VerifyIPVLANConnectivityOnSameNode() {
+	verifySRIOVConnectivity(
+		RDSCoreConfig.IPVlanNSOne,
+		RDSCoreConfig.IPVlanNSOne,
+		ipvlanDeploy20Label,
+		ipvlanDeploy21Label,
+		RDSCoreConfig.IPVlanDeploy3TargetAddress)
+
+	verifySRIOVConnectivity(
+		RDSCoreConfig.IPVlanNSOne,
+		RDSCoreConfig.IPVlanNSOne,
+		ipvlanDeploy21Label,
+		ipvlanDeploy20Label,
+		RDSCoreConfig.IPVlanDeploy4TargetAddress)
+
+	verifySRIOVConnectivity(
+		RDSCoreConfig.IPVlanNSOne,
+		RDSCoreConfig.IPVlanNSOne,
+		ipvlanDeploy20Label,
+		ipvlanDeploy21Label,
+		RDSCoreConfig.IPVlanDeploy3TargetAddressIPv6)
+
+	verifySRIOVConnectivity(
+		RDSCoreConfig.IPVlanNSOne,
+		RDSCoreConfig.IPVlanNSOne,
+		ipvlanDeploy21Label,
+		ipvlanDeploy20Label,
+		RDSCoreConfig.IPVlanDeploy4TargetAddressIPv6)
+}
+
+func defineIPVlanDeployment(dName, nsName, dLabels, netDefName, volName string,
+	dContainer *v1.Container,
+	nodeSelector map[string]string) *deployment.Builder {
+	By("Defining deployment configuration")
+
+	deploy := deployment.NewBuilder(APIClient,
+		dName,
+		nsName,
+		map[string]string{strings.Split(dLabels, "=")[0]: strings.Split(dLabels, "=")[1]},
+		dContainer)
+
+	By("Adding IPVLAN annotations")
+
+	var networks []*multus.NetworkSelectionElement
+
+	networks = append(networks,
+		&multus.NetworkSelectionElement{
+			Name:      netDefName,
+			Namespace: nsName})
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("IPVlan networks:\n\t%#v", networks)
+
+	deploy = deploy.WithSecondaryNetwork(networks)
+
+	if len(nodeSelector) > 0 {
+		By("Adding NodeSelector to the deployment")
+
+		deploy = deploy.WithNodeSelector(nodeSelector)
+	}
+
+	By("Adding Volume to the deployment")
+
+	volMode := new(int32)
+	*volMode = 511
+
+	volDefinition := v1.Volume{
+		Name: "configs",
+		VolumeSource: v1.VolumeSource{
+			ConfigMap: &v1.ConfigMapVolumeSource{
+				DefaultMode: volMode,
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: volName,
+				},
+			},
+		},
+	}
+
+	deploy = deploy.WithVolume(volDefinition)
+
+	By("Setting Replicas count")
+
+	deploy = deploy.WithReplicas(int32(1))
+
+	return deploy
+}
+
+// VerifyIPVLANSuite container that contains tests for IPVLAN verification.
+func VerifyIPVLANSuite() {
+	Describe(
+		"IPVLAN verification",
+		Label("ipvlan-clean-cluster"), func() {
+			It("Verify IPVLAN", Label("validate-new-ipvlan-different-nodes"), reportxml.ID("75057"),
+				VerifyIPVlanOnDifferentNodes)
+
+			It("Verify IPVLAN", Label("validate-new-ipvlan-same-node"), reportxml.ID("75562"),
+				VerifyIPVlanOnSameNode)
+		})
```

Comment:
I'd rather move it to the `00_validate_top_level.go` file but leaving it up to U